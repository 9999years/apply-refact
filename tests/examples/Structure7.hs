foo x = yes x x where yes x y = if a then b else if c then d else e