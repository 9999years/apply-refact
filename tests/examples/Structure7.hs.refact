[("tests/examples/Structure7.hs:1:23-67: Suggestion: Use guards\nFound:\n  yes x y = if a then b else if c then d else e\nPerhaps:\n  yes x y\n    | a = b\n    | c = d\n    | otherwise = e\n",[Replace {rtype = Match, pos = SrcSpan {startLine = 1, startCol = 23, endLine = 1, endCol = 68}, subts = [("p1001",SrcSpan {startLine = 1, startCol = 27, endLine = 1, endCol = 28}),("p1002",SrcSpan {startLine = 1, startCol = 29, endLine = 1, endCol = 30}),("g1001",SrcSpan {startLine = 1, startCol = 36, endLine = 1, endCol = 37}),("g1002",SrcSpan {startLine = 1, startCol = 53, endLine = 1, endCol = 54}),("e1001",SrcSpan {startLine = 1, startCol = 43, endLine = 1, endCol = 44}),("e1002",SrcSpan {startLine = 1, startCol = 60, endLine = 1, endCol = 61}),("e1003",SrcSpan {startLine = 1, startCol = 67, endLine = 1, endCol = 68})], orig = "yes p1001 p1002\n  | g1001 = e1001\n  | g1002 = e1002\n  | otherwise = e1003"}])]
