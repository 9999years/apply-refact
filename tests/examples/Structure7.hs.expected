foo x = yes x x where yes x y
                        | a = b
                        | c = d
                        | otherwise = e
